### PR TITLE
Provide required call flags and allowed triggers in interop descriptions.

### LINKED ISSUE #1029: Add trigger type and call flags filters to interop descriptions
And appropriate constraints on execution. We're missing `TriggerType allowedTriggers, CallFlags requiredCallFlags` in our `interop.Function`. The first one also allows to remove trigger checks from interop implementations, so we should do that too.

### PR DESCRIPTION
**Pull request by fyrchik**
Closes #1029 .

### CODE REVIEW

File: pkg/rpc/server/server.go [hunk 1]
```diff
@@ -524,6 +524,7 @@ func (s *Server) getNEP5Balances(ps request.Params) (interface{}, *response.Erro
 		cache := make(map[util.Uint160]int64)
 		for h, bal := range as.Trackers {
 			dec, err := s.getDecimals(h, cache)
+			fmt.Println(err)
```
[the remark this answers is not in the document]
Comment (fyrchik) [reply pkg/rpc/server/server.go:527]: Fixed.

File: pkg/vm/context.go [hunk 2]
```diff
@@ -41,7 +42,7 @@ type Context struct {
 	scriptHash util.Uint160
 
 	// Whether it's allowed to make dynamic calls from this context.
```
[the remark this answers is not in the document]
Comment (fyrchik) [reply pkg/vm/context.go:44]: Fixed.

File: pkg/vm/vm.go [hunk 3]
```diff
@@ -256,22 +257,18 @@ func (v *VM) Load(prog []byte) {
...
 	ctx.estack = v.estack
 	ctx.astack = v.astack
+	ctx.callFlag = f
 	v.istack.PushVal(ctx)
 }
 
-// loadScriptWithHash if similar to the LoadScript method, but it also loads
-// given script hash directly into the Context to avoid its recalculations. It's
-// up to user of this function to make sure the script and hash match each other.
-func (v *VM) loadScriptWithHash(b []byte, hash util.Uint160, hasDynamicInvoke bool) {
```
[the remark this answers is not in the document]
Comment (fyrchik) [reply pkg/vm/vm.go:268]: Fixed.

**Comment by roman-khimov** [comment]
This one needs to be rebased now.

**Pull request merged**